superdesk/factory/app.py

```diff
@@ -123,6 +123,25 @@ def init_indexes(self, ignore_duplicate_keys=False):
                     if not ignore_duplicate_keys:
                         raise
 
+    def item_context(self, name, schema=None):
+        """Register item context."""
+        self.config.setdefault("item_context", {})[name] = {
+            "schema": schema,
+        }
+
+        def update_resource_schema(resource):
```

Comment:
What are your thoughts about having a check here to make sure a new `scope` cannot override existing schema fields, only add to them?